src/read_utils.rs

```diff
@@ -100,7 +100,8 @@ pub fn get_validity_period(
 
 pub trait FileHandler {
     type Reader: std::io::Read;
-    fn get_file(self, name: &str) -> Result<Self::Reader>;
+
+    fn get_file(self, name: &str) -> Result<(Option<Self::Reader>, PathBuf)>;
```

Comment:
I know that's only 3 copy/paste for the moment, but it might be more later

```rust
fn get_file_if_exists(self, name: &str) -> Result<(Option<Self::Reader>, PathBuf)>;
fn get_file(self, name: &str) -> Result<(Self::Reader, PathBuf)> {
    let (reader, path) = self.get_file_if_exists(name)?;
    Ok((reader.ok_or_else(|| format_err!("file {:?} not found", path))?, path)
}
```